packages/components/src/button/index.tsx

```diff
@@ -171,9 +190,30 @@ export function Button( props, ref ) {
 				<Icon icon={ icon } size={ iconSize } />
 			) }
 			{ children }
-		</Tag>
+		</>
 	);
 
+	const element =
+		Tag === 'a' ? (
+			<a
+				{ ...( elementProps as WordPressComponentProps<
+					ButtonProps,
+					'a'
+				> ) }
+			>
+				{ elementChildren }
+			</a>
+		) : (
+			<button
+				{ ...( elementProps as WordPressComponentProps<
+					ButtonProps,
+					'button'
+				> ) }
+			>
+				{ elementChildren }
+			</button>
+		);
```

Comment:
This is really verbose.

But without this, there are type errors:

```
const Tag: "a" | "button"
Type '{ children: (ReactNode | Element)[]; className: string; "aria-label": string | undefined; "aria-describedby": string | undefined; ref: ForwardedRef<any>; ... 356 more ...; wrap?: string | undefined; }' is not assignable to type 'AnchorHTMLAttributes<HTMLAnchorElement>'.
  Types of property 'onCopy' are incompatible.
    Type 'ClipboardEventHandler<HTMLAnchorElement> | ClipboardEventHandler<HTMLButtonElement> | ClipboardEventHandler<...> | undefined' is not assignable to type 
```

![Screen Shot 2023-01-15 at 4 25 28 PM](https://user-images.githubusercontent.com/4063887/212570500-da013218-bbde-42e2-aac3-85b3a463f38b.png)